pkg/encoding/v1/encoding.go

```diff
@@ -481,5 +553,47 @@ func (d *Decoder) Decode(data []byte) (*object.OpenCompose, error) {
 		openCompose.Volumes = append(openCompose.Volumes, ov)
 	}
 
+	// convert secrets
+	for _, secret := range v1.Secrets {
+		var os object.Secret
+		os.Name = string(secret.Name)
+		for _, secData := range secret.Data {
+
+			// If the secret has been provided through a file
+			// and
+			// if it is a relative path
+			// and
+			// the OpenCompose input file is either STDIN or file path
+			// then
+			// convert it to an absolute file path.
+			// This will fail if input file is URL
+			if (secData.File != nil) && !filepath.IsAbs(*secData.File) {
+
+				if in.URL != nil {
+					return nil, fmt.Errorf("Unable to decode secret %v with key %v has a relative file path %v, but is being passed as a URL", secret.Name, secData.Key, secData.File)
```

Comment:
We can start with just rewording error message. Loading files from local filesystem can be done later.